k-distribution/tests/regression-new/map-symbolic-tests-haskell/remove-9-spec.k.out

```diff
@@ -25,6 +25,12 @@
   #Equals
     X:MyId in_keys ( MAP )
   }
+#And
+  {
+    false
+  #Equals
+    X:MyId in_keys ( MAP [ Y:MyId <- undef ] )
```

Comment:
Can we add rules to simplify this more? (If so, we can do it in another pull request.)